src/networking/messageHandler.ts

```diff
@@ -1,31 +1,59 @@
 import { messageTracker } from "./messageTracker";
 import { Stats, writeFile } from "fs";
+import { RawData } from "ws";
 import { config } from "../config";
 import { EventType } from "../eventTypes";
 import { fileChangeEventToMsg } from "./messageGenerators";
 import type { Signal } from "signal-js";
 import { Message } from "../interfaces";
 
-export function messageHandler(signaller: Signal, msg: Message, paths: Map<string, Stats>) {
-  let incoming;
+function deserialize(data: RawData): Message | void {
```

Comment:
Instead of this returning void, I think it'd make more sense to return only `Message`, have the errors actually throw, then handle messaging wherever this is called. Especially considering there's _also_ error-handling there.

Reply:
I just made a commit that should do what you asked for, please let me know if it needs further changes.